### PR TITLE
My tasks endpoint

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

File: lib/routers/profile/pil.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const router = Router({ mergeParams: true });
 const submit = (action) => {
   return (req, res, next) => {
     const params = {
-      action,
+      action: action || req.params.action,
```
Comment (joefitter) [inline lib/routers/profile/pil.js]: hmmm, this means any action will be passed to workflow, what if the action is `delete`?

File: lib/routers/user.js [hunk 2]
```diff
@@ -18,5 +18,6 @@ router.use((req, res, next) => {
 });
 
 router.use(require('./profile'));
+router.use('/tasks', require('./tasks'));
```
Comment (joefitter) [inline lib/routers/user.js]: do we ever want to be able to get _another_ user's tasks? if so we should mount in profile router

File: lib/routers/profile/pil.js (hunk 1 above)
Comment (wheelsandcogs) [reply lib/routers/profile/pil.js]: I was supposed to be applying a check at some point - good catch

File: lib/routers/user.js (hunk 2 above)
Comment (wheelsandcogs) [reply lib/routers/user.js]: As I understand it - if a task is relevant to you it should be in your task list - I'm not sure if viewing other peoples task lists is a thing?

Comment (joefitter) [reply lib/routers/user.js]: I am also not sure if it is a thing. I was more thinking of a super user having visibility over tasks assigned to a different user, and using permissions middleware to decide if they are able to request tasks for the given profile

Comment (wheelsandcogs) [reply lib/routers/user.js]: ok let's change it later if the requirement to do that comes up.

File: lib/routers/profile/pil.js (hunk 1 above)
Comment (wheelsandcogs) [reply lib/routers/profile/pil.js]: added a `validateAction` for now - we probably want to tie this to user perms somehow but I need to discuss with you lot.

File: lib/routers/profile/pil.js [hunk 3]
```diff
@@ -35,6 +35,16 @@ const validate = (req, res, next) => {
   })(req, res, next);
 };
 
+const validateAction = (req, res, next) => {
+  const pilActions = ['grant', 'endorse', 'revoke'];
+
+  if (!pilActions.includes(req.params.action)) {
+    throw Error('unrecognised action');
```
Comment (joefitter) [inline lib/routers/profile/pil.js]: `Error` is a constructor so should be called with `new`

**Pull request merged**